File: AlgoExpert/python/two_number_sum.py
"""
Difficulty: Easy
"""

from typing import List


def twoNumberSum(array: List[int], targetSum: int) -> List[int]:
    # Write your code here.
    pairs = set()

    for num in array:
        if num in pairs:
            return [num, targetSum - num]
        pairs.add(targetSum - num)

    return []

File: AlgoExpert/cpp/two_number_sum.cc
/* Notes
Difficulty: Easy
*/

#include <unordered_set>
#include <vector>

std::vector<int> twoNumberSum(const std::vector<int> array,
                              const int targetSum) {
  // Write your code here.
  std::unordered_set<int> pairs;

  for (const int num : array) {
    if (pairs.find(num) != pairs.end()) {
      return {num, targetSum - num};
    }
    pairs.insert(targetSum - num);
  }
  return {};
}